src/CustomerList/index.js

```diff
@@ -3,63 +3,57 @@ import './index.css'
 import Header from '../components/Header';
 import Count from './Count/index';
 import TableList from '../components/TableList';
-
+import { fetchUsers } from '../actions/user';
+import { connect } from 'react-redux';
 
 class CustomerList extends React.Component {
     constructor(props) {
         super(props);
-        this.state = {
-          error: "",
-          list: []
-        };
-      }
-    
-      componentDidMount() {
-        fetch('/table-data',{
-          method:"GET",
-          headers: {
-            'Content-Type': 'application/json',
-          },
-        })
-        .then(res => res.json())
-        .then(result => {
-          this.setState({
-            list: result
-          })
-        })
-        .catch(error => 
-          this.setState({
-            error: "somthing went wrong"
-          })
-        );
-      }
-    
-      render() {
-            const { error,list } = this.state;
-            if(error){
-                  console.log(error);
-                  return <div>Error: {error}</div>;
-            }
-           
-            return(
-                    <>
-                        <Header />
-                        <Count />
-                        <TableList 
-                          headers={[
-                            {key:"id",display:"ID"},
-                            {key:"name",display:"NAME"},
-                            {key:"address",display:"ADDRESS"},
-                            {key:"Email",display:"EMAIL"},
-                            {key:"password",display:"PASSWORD"}
-                          ]} 
-                          list={list}
-                        />
-                     </>
-                  )
-        }
+        this.state = {}
+
     }
+    componentDidMount() {
+        this.props.fetchUsers();
+       
+    }
+    render() {
+          const { error,list } = this.state;
+          if(error){
+                return <div>Error: {error}</div>;
+          }
+          const userData = this.props.user.user_list;
+          return(
+                  <>
+                      <Header />
+                      <Count />
+                      {/* <h2>{JSON.stringify(this.props.list.users[0])}</h2> */}
+                      <TableList 
+                        headers={[
+                          {key:"id",display:"ID"},
+                          {key:"name",display:"NAME"},
+                          {key:"address",display:"ADDRESS"},
+                          {key:"Email",display:"EMAIL"},
+                          {key:"password",display:"PASSWORD"}
+                        ]} 
+                        list={userData}
+                      />
+                    </>
+                )
+      }
+}
+
+const mapStateToProps = state => {
+  return {
+    user: state.user,
+    error: state.user.error
```

Comment:
now user state has all -> list, error, loading. So no need to have a separate access 
`error: state.user.error`